text/language-bindings/README.md

```diff
@@ -0,0 +1,196 @@
+Thin wrappers should be included in the Ursa project itself in a folder called `wrappers`. This will allow them to
+stay in-sync with any other changes to the core library.
+
+Idiomatic wrappers usually cannot be generated by hand as they require more in depth language experience
+than automatic generators can produce. Therefore, idiomatic wrappers should be written by project contributors
+and maintained in separate repositories. These wrappers also SHOULD include tests for interacting with
+native Ursa.
+
+Here are general guidelines for producing wrappers:
+
+`<language>` should follow the naming convention as described [here](https://support.codebasehq.com/articles/tips-tricks/syntax-highlighting-in-markdown)
+
+- Thin wrappers
+    - Produced by code
+    - Output into the `libursa/wrappers/<language>/folder`
+- Idiomatic wrappers
+    - Written by developers
+    - Implemented as best deemed by contributors/community
+    - Stored in separate repositories using the naming convention `ursa-wrapper-<language>`
+    - Keep certain things invariant
+        - The reason is to make wrappers similar to one another, and to avoid unnecessary documentation burden or mental model translation between wrappers (which may have light doc) and ursa (where the investment in doc will be substantial).
+        - If someone learns how to use a wrapper of the API in language 1, and then go look at the wrapper of the API in language 2, one should see the same basic concepts, and be able to predict the function names one should call and the sequence of calls one should use.
+        - Each wrapper should preserve all the terminology of the main interface: don't change `encrypt` to `seal` or `decrypt` to `open`
+        - All preconditions on parameters, all data formats (e.g., json, config files, etc), all constraints on input should be identical in all wrappers and in the base API.
+        - The "level" of the wrapper API can be generally "higher" than the C-callable functions, but it should expose the same amount of functionality and flexibility as the C-callable functions, so the higher level functions, if they exist, should be an addition to lower-level ones, not substitutes for them. In other words, helpers and convenience methods are great, but don’t allow them to entirely hide the core functions which provide granular control.
+    - The wrapper should document the earliest and latest version of ursa that it knows to be compatible.
+    - The wrapper should document what platforms it targets, what use cases it's built for, etc. A wrapper should be able to find ursa using default OS methods (e.g., in the system PATH), but should also provide a way for a specific path to ursa to be specified, such that the wrapper can work either from an OS-wide install of ursa or a version in a particular directory.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+- Some developers may want all ursa code written in the language of their needs to avoid issues presented by FFI.
```

Comment:
Maybe also mention how this relates to signing in-browser with Javascript. (Probably via wasm?)

Reply:
So you want an example?